### PR TITLE
Handle incoming s3 delete events

### PR DESCRIPTION
**Pull request by spacid**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -224,10 +216,139 @@ def callback(ch, method, properties, body, ctx):
 
     events = Events(ctx.config.app_cfg["rabbitmq"]["outgoing"], ctx)
     events.publish(json.dumps(param_dict), properties.correlation_id)
+    return True
+
 
+def delete_media_object(mediahaven_service, fragment_id: str, reason: str) -> bool:
```
Comment (maartends) [inline main.py:222]: Should better be a method on the MediaHaven-svc.
And a bit awkwardly placed in between the create and the remove handler.

File: main.py [hunk 2]
```diff
@@ -114,29 +112,24 @@ def construct_collateral_sidecar(event, pid, media_id):
...
-        event = json.loads(body)
-    except json.JSONDecodeError as error:
-        log.warning("Bad s3 event.", error=str(error))
-        return
 
+def create_handler(event: dict, properties, ctx: Context) -> bool:
+    """Handler for s3 create events"""
     # Check if item already in mediahaven based on key and md5
     mediahaven_service = MediahavenService(ctx)
     query_params = [
```
Comment (maartends) [inline main.py:135]: We need to add the bucket as well: new ticket.

File: main.py [hunk 3]
```diff
@@ -224,10 +216,139 @@ def callback(ch, method, properties, body, ctx):
...
+        # Delete the collaterals
+        for fragment_collateral in fragments_collateral:
+            # Get the Fragment ID of the fragment to which this collateral is linked to
+            media_id = fragment_collateral[1]
+            linked_fragment_id = fragments.get(media_id)
+            result = delete_media_object(
+                mediahaven_service,
+                fragment_collateral[0],
+                f'This was a collateral with Media ID: "{media_id}" linked to fragment with Fragment ID: "{linked_fragment_id}". Underlying essence has been deleted via s3 delete-object'
+            )
```
Comment (maartends) [inline main.py]: For VRT, this is a "Media ID". Generally speaking, however, this is a `local_id`.

```suggestion
            local_id = fragment_collateral[1]
            linked_fragment_id = fragments.get(local_id)
            result = delete_media_object(
                mediahaven_service,
                fragment_collateral[0],
                f'Deleted collateral with local_id: "{local_id}" linked to fragment with fragment_id: "{linked_fragment_id}". Essence was deleted via s3 delete-object.'
            )
```

**Review by maartends — CHANGES_REQUESTED** [review]
### Handler naming

Ambiguity exists between `get_handler` on one hand, and `create_handler` and `remove_handler` on the other hand. By naming alone, they seem to to "handle" three different actions: get, create and delete. Which is not the case.

File: main.py (hunk 1 above)
Comment (spacid) [reply main.py:222]: The actual delete method is in MH-svc. This method logs and has some error handling.
It is placed just above the method that handles the delete events, as that one uses this logic.

Shall I move it above the create handler?

**Review by RudolfDG — APPROVED** [review]

**Review by maartends — APPROVED** [review]

**Pull request merged**